runtime/compiler/z/codegen/J9TreeEvaluator.cpp

```diff
@@ -211,6 +211,372 @@ doubleMaxMinHelper(TR::Node *node, TR::CodeGenerator *cg, bool isMaxOp)
+   regDeps->addPostCondition(r_tmp            , TR::RealRegister::AssignAny);
+   regDeps->addPostCondition(r_tmp2           , TR::RealRegister::AssignAny);
+   regDeps->addPostCondition(r_1stCharIndex   , TR::RealRegister::AssignAny);
+
+   regDeps->addPostCondition(v_s1Partial , TR::RealRegister::AssignAny);
+   regDeps->addPostCondition(v_s2Partial , TR::RealRegister::AssignAny);
+   regDeps->addPostCondition(v_s2Char1Rep, TR::RealRegister::AssignAny);
+   regDeps->addPostCondition(v_tmp       , TR::RealRegister::AssignAny);
+
+   // Labels
+   TR::LabelSymbol* label_find_1st_char        = TR::LabelSymbol::create(cg->trHeapMemory(), cg);
+   TR::LabelSymbol* label_load_len_16          = TR::LabelSymbol::create(cg->trHeapMemory(), cg);
+   TR::LabelSymbol* label_load_len_done        = TR::LabelSymbol::create(cg->trHeapMemory(), cg);
+   TR::LabelSymbol* label_extract_1st_char_pos = TR::LabelSymbol::create(cg->trHeapMemory(), cg);
+   TR::LabelSymbol* label_match_s2_loop        = TR::LabelSymbol::create(cg->trHeapMemory(), cg);
+   TR::LabelSymbol* label_match_s2_loop_prep   = TR::LabelSymbol::create(cg->trHeapMemory(), cg);
+   TR::LabelSymbol* label_s2_partial_match     = TR::LabelSymbol::create(cg->trHeapMemory(), cg);
+   TR::LabelSymbol* label_load_result          = TR::LabelSymbol::create(cg->trHeapMemory(), cg);
+   TR::LabelSymbol* label_done                 = TR::LabelSymbol::create(cg->trHeapMemory(), cg);
+
+   // Ops
+   TR::InstOpCode::Mnemonic AHI  = is64Bit ? TR::InstOpCode::AGHI : TR::InstOpCode::AHI;
+   TR::InstOpCode::Mnemonic AHIK = is64Bit ? TR::InstOpCode::AGHIK : TR::InstOpCode::AHIK;
+   TR::InstOpCode::Mnemonic ARK  = is64Bit ? TR::InstOpCode::AGRK : TR::InstOpCode::ARK;
+   TR::InstOpCode::Mnemonic AR   = is64Bit ? TR::InstOpCode::AGR : TR::InstOpCode::AR;
+   TR::InstOpCode::Mnemonic CRJ  = is64Bit ? TR::InstOpCode::CGRJ : TR::InstOpCode::CRJ;
+   TR::InstOpCode::Mnemonic CIJ  = is64Bit ? TR::InstOpCode::CGIJ : TR::InstOpCode::CIJ;
+   TR::InstOpCode::Mnemonic LHI  = is64Bit ? TR::InstOpCode::LGHI : TR::InstOpCode::LHI;
+   TR::InstOpCode::Mnemonic LR   = is64Bit ? TR::InstOpCode::LGR : TR::InstOpCode::LR;
+   TR::InstOpCode::Mnemonic SLL  = is64Bit ? TR::InstOpCode::SLLG : TR::InstOpCode::SLL;
+   TR::InstOpCode::Mnemonic SRK  = is64Bit ? TR::InstOpCode::SGRK : TR::InstOpCode::SRK;
+
+   // Start
+   TR::Instruction* cursor = generateRIInstruction(cg, TR::InstOpCode::LHI, node, r_resultReg, -1);
+   generateRRInstruction(cg, TR::InstOpCode::LGFR, node, r_s1Len, r_s1Len);
+   generateRRInstruction(cg, TR::InstOpCode::LGFR, node, r_s2Len, r_s2Len);
+   generateRRInstruction(cg, TR::InstOpCode::LGFR, node, r_fromIndex, r_fromIndex);
+
+   if (isUTF16)
+      {
+      generateRSInstruction(cg, SLL, node, r_s1Len, r_s1Len, 1);         // these are byte lengths
+      generateRSInstruction(cg, SLL, node, r_s2Len, r_s2Len, 1);
+      generateRSInstruction(cg, SLL, node, r_fromIndex, r_fromIndex, 1);
```

Comment:
Can't we use `RISBG` instructions here to do the `LGFR` + `SLL` all in one shot? Search around the codebase there is an API to do this. @dchopra001 should be familiar with it if you need help.

Reply:
Updated the code to use `generateShiftAndKeepSelected64Bit` to generate `RISBGs`.